### ISSUE TITLE
[a11y-defect-2]: Edit Benefit Letters update address buttons must be contextual for assistive technology users

### ISSUE THREAD

**Issue by 1Copenut** [department-of-veterans-affairs/va.gov-team#96346]
### Point of contact

Trevor Pierce

### Severity level

2, Serious. Should be fixed in 1-2 sprints post-launch.

### Details

When Benefit Letter users update their address, we have a secondary routine that asks users to confirm their address. They can either use the address as entered or use the USPS recommended address. In this routine, the primary action button says "Use this address" but non-sighted users have no context what "this" means in this case. It could lead to a situation where errors are introduced or addresses incorrectly updated. Screenshot attached.

---
![Image](https://github.com/user-attachments/assets/6e7baff7-6dcf-4350-9012-34e8088ecf8f)

### Reproduction steps

1. Log into https://staging.va.gov/records/download-va-letters/letters/confirm-address with a test user who has benefit letters. I used 54.
1. Turn on a screen reader of your choosing
1. Click the "Edit" button to switch the mailing address to the inline form
2. Edit or change one or two fields in the form
3. Press the "Save" button
4. Verify the updated alert box announces itself to your screen reader
5. Tab to the Use this address button and consider how important it is to know if you're using the suggested or as entered address

### Proposed solution or next steps

The button text should be more descriptive and should be contextual to which radio button users have selected. That way they have a chance to know what address is being used and double check as they feel necessary.

### References, articles, or WCAG support

1. Understanding [SC 1.3.1](https://w3.org/TR/WCAG22#info-and-relationships): Info and Relationships (Level A)
2. Understanding [SC 3.3.4](https://w3.org/TR/WCAG22#error-prevention-legal-financial-data): Error Prevention (Legal, Financial, Data) (Level AA)

### Type of issue

- [ ] Axe-core
- [x] Screenreader | assistive tech & device support
- [ ] Keyboard
- [ ] Zoom
- [ ] Color, typography, & visual elements
- [ ] Components and pattern standards
- [ ] Content organization
- [ ] Focus
- [ ] Headings
- [ ] Content style guide
- [x] Link and button labels
- [ ] Semantic HTML
- [ ] Markup and metadata
- [ ] Non-text content and media
- [ ] Overall user experience
- [x] Cognitive | user flows & navigating
- [ ] URL standards
- [ ] Something else

**Comment by dysbo** [comment]
Discussed in planning and we love it, let's go 👍

**Comment by jstrothman** [comment]
All of @rallencf 's suggested changes sound good to me.
Could you also:
1. Change the form label from `You entered` to `Address you entered`
2. Make sure the form label `Suggested addresses` is only plural if we're suggesting more than one address

**Comment by jstrothman** [comment]
@dysbo @1Copenut @rallencf @sh0wn On second thought, what do you think about making this interaction more form-like:

1. Move the form instruction out of the alert and change it from "Tell us which of these addresses you'd like us to use" to "Which of these addresses should we use?" (Marked up as a heading -- whatever's the next level heading under "Mailing address"
2. Use the [`tile` style of radio buttons](https://design.va.gov/components/form/radio-button#tile) for the address options

**Comment by dysbo** [comment]
@jstrothman I love the tile radio buttons for this.  I would love to get everyone's thoughts.  Also @1Copenut and @janechodance if we want to reevaluate the points we put on this after we land on a direction, let's do that 👍

**Comment by rallencf** [comment]
@jstrothman Good call. I like pulling the instructions out of the alert box—and that text change. More straightforward.

**Comment by dysbo** [comment]
If we could get a quick Figma or similar of what we want the updates to look like, that'd be stellar 🙏 @sh0wn I guess is who I tag? if you have a chance 🙂

**Comment by sh0wn** [comment]
@dysbo @jstrothman Happy to sketch something out today

**Comment by dysbo** [comment]
@1Copenut is going to take a look at this and see what our options are.  We'll circle back around Monday to see what's possible and what we can achieve.
